ambry-store/src/main/java/com.github.ambry.store/StoreManager.java

```diff
@@ -71,37 +98,75 @@ We do this by making sure that the retention period for deleted messages (which
     }
   }
 
+  /**
+   * Start the {@link DiskManager}s for all disks on this node.
+   */
   public void start()
-      throws StoreException {
+      throws InterruptedException {
     logger.info("Starting store manager");
-    // iterate through the replicas for this node and create the stores
-    for (ReplicaId replica : replicas) {
-      // check if mount path exist
-      File file = new File(replica.getMountPath());
-      if (!file.exists()) {
-        throw new IllegalStateException("Mount path does not exist " + replica.getMountPath());
-      }
-      // Partition id is used as the id of a store.
-      String storeId = replica.getPartitionId().toString();
-      Store store =
-          new BlobStore(storeId, config, scheduler, registry, replica.getReplicaPath(), replica.getCapacityInBytes(),
-              factory, recovery, hardDelete, time);
-      store.start();
-      stores.put(replica.getPartitionId(), store);
+    List<Thread> startupThreads = new ArrayList<>();
+    for (final DiskManager diskManager : diskManagers) {
+      Thread thread = Utils.newThread("disk-manager-startup-" + diskManager.getDisk(), new Runnable() {
+        @Override
+        public void run() {
+          try {
+            diskManager.start();
+          } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            logger.error("Disk manager startup thread interrupted for disk " + diskManager.getDisk(), e);
+          }
+        }
+      }, false);
+      thread.start();
+      startupThreads.add(thread);
+    }
+    for (Thread startupThread : startupThreads) {
+      startupThread.join();
     }
     logger.info("Starting store manager complete");
   }
 
+  /**
+   * @param id the {@link PartitionId} to find the store for.
+   * @return the {@link Store} corresponding to the given {@link PartitionId}, or {@code null} if no store was found for
+   *         that partition, or that store was not started.
+   */
   public Store getStore(PartitionId id) {
-    return stores.get(id);
+    DiskManager diskManager = partitionToDiskManager.get(id);
+    return diskManager != null ? diskManager.getStore(id) : null;
```

Comment:
Perhaps you should treat it as an error if the `StoreManager` is asked for a `Store` that it does not manage, and throw a `StoreException` (currently this would end up as a `NullPointerException` somewhere).